lega/utils/storage.py

```diff
@@ -231,8 +240,7 @@ def open(self, path, mode='rb'):
 
     def exists(self, path):
         """Return true if the path exists."""
-        resp = self.s3.head_object(Bucket=self.bucket, Key=path)
-        return bool(resp['ContentLength'])
+        return bool(self.filesize(path))
```

Comment:
I have not tested, but what if the file does exist but is empty, what will be the return value?

Reply:
It will return 0 in that case as the ContentLength. The `bool` will make the `exists` return `false`.

Reply:
So an empty file is considered as "NonExist". Seems not that logical